HealthcareSystem/Backend/Service/DrugAndTherapy/TherapyService.cs

```diff
@@ -75,5 +75,126 @@ public void AddTherapy(Therapy therapy)
         {
             _therapyRepository.AddTherapy(therapy);
         }
+
+        public List<Therapy> GetTherapiesByExaminationSearch(List<Therapy> therapies, string startDate, string endDate, string doctorSurname, string drugName)
```

Comment:
A method contains too many lines of code. Generally, any method longer than ten lines should make you start asking questions.

If you feel the need to comment on something inside a method, you should take this code and put it in a new method. Even a single line can and should be split off into a separate method, if it requires explanations. And if the method has a descriptive name, nobody will need to look at the code to see what it does.

To reduce the length of a method body, use Extract Method ( https://refactoring.guru/extract-method ).